src/validation.cpp

```diff
@@ -1016,31 +1016,45 @@ bool GetTransaction(const uint256& hash, CTransactionRef& txOut, const Consensus
         CTransactionRef ptx = mempool.get(hash);
         if (ptx) {
             txOut = ptx;
-            return true;
+            return GetTransactionResult::LOAD_OK;
         }
 
         if (fTxIndex) {
             CDiskTxPos postx;
             if (pblocktree->ReadTxIndex(hash, postx)) {
+                if (postx.nFile == std::numeric_limits<int>::max()) {
+                    // blockfile is not available (pruned), CDiskTxPos.nPos points to the blockheight
+                    // lookup blockindex via height from txindex and set hashBlock
+                    CBlockIndex *p_index_possible = chainActive[postx.nPos];
+                    if (p_index_possible) {
+                        hashBlock = *p_index_possible->phashBlock;
```

Comment:
`hashBlock` is only updated if block index is found. Either `assert(hashBlock.IsNull())` before or just call `hashBlock.SetNull()`.